homeassistant/components/powerwall/__init__.py

```diff
@@ -116,14 +130,37 @@ async def _update_data(self) -> PowerwallData:
                 return data
         raise RuntimeError("unreachable")
 
+    async def save_auth_cookie(self) -> None:
```

Comment:
```suggestion
    @callback
    def async_save_auth_cookie(self) -> None:
```

Nothing is being awaited here, this can be marked as an async safe callback function instead and the `await` removed